pkg/app/cockroachdb.go

```diff
@@ -0,0 +1,303 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/helm"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/kanisterio/kanister/pkg/poll"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const cReadyTimeout = 10 * time.Minute
+
+type CockroachDB struct {
+	name      string
+	namespace string
+	cacrt     string
+	tlscrt    string
+	tlskey    string
+	cli       kubernetes.Interface
+	chart     helm.ChartInfo
+}
+
+// NewCockroachDB Last tested working version "22.1.5"
+func NewCockroachDB(name string) App {
+	return &CockroachDB{
+		name: name,
+		chart: helm.ChartInfo{
+			Release:  appendRandString(name),
+			RepoName: helm.CockroachDBRepoName,
+			RepoURL:  helm.CockroachDBRepoURL,
+			Chart:    "cockroachdb",
+			Values: map[string]string{
+				"image.tag":        "v22.1.5",
+				"image.pullPolicy": "IfNotPresent",
+			},
+		},
+	}
+}
+
+func (c *CockroachDB) Init(context.Context) error {
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+	c.cli, err = kubernetes.NewForConfig(cfg)
+	return err
+}
+
+func (c *CockroachDB) Install(ctx context.Context, namespace string) error {
+	log.Info().Print("Installing cockroachdb cluster helm chart.", field.M{"app": c.name})
+	c.namespace = namespace
+
+	// Create the helm client
+	cli, err := helm.NewCliClient()
+	if err != nil {
+		return errors.Wrap(err, "failed to create helm client")
+	}
+
+	// Add helm repo and fetch charts
+	if err = cli.AddRepo(ctx, c.chart.RepoName, c.chart.RepoURL); err != nil {
+		return errors.Wrapf(err, "Failed to install helm repo. app=%s repo=%s", c.name, c.chart.RepoName)
+	}
+
+	// Install c operator, admission controller and cluster
+	err = cli.Install(ctx, fmt.Sprintf("%s/%s", c.chart.RepoName, c.chart.Chart), c.chart.Version, c.chart.Release, c.namespace, c.chart.Values)
+	//time.Sleep(5 * time.Minute)
+	return errors.Wrapf(err, "Failed to install helm chart. app=%s chart=%s release=%s", c.name, c.chart.Chart, c.chart.Release)
+}
+
+func (c *CockroachDB) IsReady(ctx context.Context) (bool, error) {
+	log.Info().Print("Waiting for cockroachdb cluster to be ready.", field.M{"app": c.name, "namespace": c.namespace, "release": c.chart.Release})
+	// Add timeout to context
+	ctx, cancel := context.WithTimeout(ctx, cReadyTimeout)
+	defer cancel()
+	err := kube.WaitOnStatefulSetReady(ctx, c.cli, c.namespace, c.chart.Release)
+	if err != nil {
+		log.WithError(err).Print("Error Occurred --> ", field.M{"error": err.Error()})
+		return false, err
+	}
+	log.Print("Application instance is ready.", field.M{"app": c.name})
```

Comment:
I see at some places we are using `log.Info.Print` and here we are just using `log.Print` let's try to be consistent and use one format. We can refer other apps to see how its done.